apps/cowswap-frontend/src/common/updaters/FeesUpdater.ts

```diff
@@ -61,27 +61,60 @@ function quoteUsingSameParameters(currentParams: FeeQuoteParams, quoteInfo: Quot
   } = currentParams
   const { amount, buyToken, sellToken, kind, userAddress, receiver, appData } = quoteInfo
   const hasSameReceiver = currentReceiver && receiver ? currentReceiver === receiver : true
+  const hasSameAppData = compareAppDataWithoutQuoteData(appData, currentAppData)
 
   // cache the base quote params without quoteInfo user address to check
   const paramsWithoutAddress =
     sellToken === currentSellToken &&
     buyToken === currentBuyToken &&
     amount === currentAmount &&
     kind === currentKind &&
-    appData === currentAppData &&
+    hasSameAppData &&
     hasSameReceiver
   // 2 checks: if there's a quoteInfo user address (meaning quote was already calculated once) and one without
   // in case user is not connected
   return userAddress ? currentUserAddress === userAddress && paramsWithoutAddress : paramsWithoutAddress
 }
 
+/**
+ * Compares appData without taking into account the `quote` metadata
+ */
+function compareAppDataWithoutQuoteData<T extends string | undefined>(a: T, b: T): boolean {
```

Comment:
Just to keep the file clean, would be better to move the new functions into a new file